### ISSUE TITLE
Regression: --with-testnic* broken as of 4.0.5

### ISSUE THREAD

**Issue by idl0r** [appneta/tcpreplay#178]
Hi,

the above mentionend options (--with-testnic=/--with-testnic2=) makes it impossible to specify any interfaces at all:

```
# tcpreplay -i vnet0 fud.pcap
Fatal Error: Unable to parse args: Invalid interface name/alias: vnet0
```

It doesn't matter what interface you pass, it always fails and throws this error.
This happens as of 4.0.5.

See also:
https://bugs.gentoo.org/show_bug.cgi?id=546762

**Comment by fklassen** [comment]
This works for me, so I suspect that there is something wrong with the version of libpcap used. Tcpreplay simply calls its `pcap_findalldevs()` function, and if the interface does not show up, it cannot test to it. If 4.0.4 worked, I suspect it was compiled to a different version of libpcap.

I have trouble understanding the resolution description for the bug reported on Gentoo Bugzilla. Am I missing something?

**Comment by idl0r** [comment]
Hi Fred,

```./configure --enable-dynamic-link --with-testnic2=lo --with-testnic=lo```
...
```
./tcpreplay -i vnet0 fud.pcap
Fatal Error: Unable to parse args: Invalid interface name/alias: vnet0
```

Without those two options it works as expected. I'm not sure if and how pcap_findalldevs() is related.

**Comment by fklassen** [comment]
These options have no effect on the source code. It only affects `make tests`, and should not otherwise affect the executables. Not sure what is going on, but will close for now.

**Comment by fklassen** [comment]
Reopen to investigate this issue and #474. Discovered that tests fail when using `--with-testnic` on Linux.

**Comment by fklassen** [comment]
Fixed in PR #503
